plugins/woocommerce/includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -131,6 +131,18 @@ private function obtain_lock_on_sku_for_concurrent_requests( $product ) {
 			$sku,
 			$sku
 		);
+
+		/**
+		 * Filters the query to lock SKU on product creation.
+		 * Expects a prepared SQL query.
+		 *
+		 * @since 9.2.0
+		 *
+		 * @param $query   The query to lock SKU.
+		 * @param $product The product being created.
+		 */
+		$query = apply_filters( 'wc_lock_sku_query', $query, $product );
```

Comment:
```suggestion
		/**
		 * Filter to bail early on the SKU lock query.
		 *
		 * @since 9.3.0
		 *
		 * @param bool|null  $locked  Set to a boolean value to short-circuit the SKU lock query.
		 * @param WC_Product $product The product being created.
		 */
		$locked = apply_filters( 'wc_product_pre_lock_on_sku', null, $product );
		if ( ! is_null( $locked ) ) {
			return boolval( $locked );
		}
```

Hi @Hug0-Drelon thanks for the PR! I understand the need for this based on your description in the issue. I do want to clarify, though, that in `wc_product_has_unique_sku()` the filters don't actually allow you to modify the query, they just allow you to modify the _result_. I think we'd want to do that in this case too. It's just safer and less error-prone if the filtered value is a boolean rather than a query string that needs to have been correctly prepared.

Reply:
Good idea, as long as there's only one query run for performance. Thanks for the prop!